src/table/persistence/QueryInterpreter/QueryNodes/NotEqualsNode.ts

```diff
@@ -0,0 +1,24 @@
+import { IQueryContext } from "../IQueryContext";
+import BinaryOperatorNode from "./BinaryOperatorNode";
+import GuidNode from "./GuidNode";
+
+export default class NotEqualsNode extends BinaryOperatorNode {
+  get name(): string {
+    return `ne`
+  }
+
+  evaluate(context: IQueryContext): any {
+    if (!this.backwardsCompatibleGuidEvaluate(context)) {
+      return false
+    }
+
+    return this.left.evaluate(context) !== this.right.evaluate(context)
+  }
+
+  private backwardsCompatibleGuidEvaluate(context: IQueryContext): boolean {
+    const left = this.left instanceof GuidNode ? this.left.stringGuid() : this.left.evaluate(context)
+    const right = this.right instanceof GuidNode ? this.right.stringGuid() : this.right.evaluate(context)
+
+    return left !== right
```

Comment:
This issue still exist: https://github.com/Azure/Azurite/issues/1929
This code change might can fix it. And the `evaluate()` function above should also be fix similarly.
```suggestion
    if (left === undefined || right === undefined)
    {
      return false;
    }
    return left !== right
```

Reply:
Thank you, I've added that improvement as well as a test case to ensure we don't regress on this.